### PR TITLE
[PREVIEW] Rdm 3063 Case Progression: Extract case progression with existing contract

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-3063

### Change description ###
Case Progression: Extract case progression with existing contract

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -119,7 +119,7 @@
     "protractor-multiple-cucumber-html-reporter-plugin": "^1.8.0",
     "protractor-screenshoter-plugin": "^0.10.1",
     "puppeteer": "^1.6.0",
-    "sonar-scanner": "^3.0.3",
+    "sonar-scanner": "^2.9.2",
```
Comment (andrewfolga) [inline package.json]: Hmm unsure whether this needed to be downgraded - will verify.

File: src/app/workbasket/workbasket.component.ts [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 import { PaginationService } from '../core/pagination/pagination.service';
-import { CaseType } from '../shared/domain/definition/case-type.model';
 import { FormGroup } from '@angular/forms';
 import { SearchResultComponent } from '../shared/search/search-result.component';
 import { Observable } from 'rxjs';
-import { AlertService } from '../core/alert/alert.service';
 import { plainToClass } from 'class-transformer';
+import { Profile, Jurisdiction, AlertService } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
+import { CaseState } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-state.model';
```
Comment (vlaurin) [inline src/app/workbasket/workbasket.component.ts]: The 2 imports above seems wrong. Shouldn't it just be `from '@hmcts/ccd-case-ui-toolkit'` ?

File: src/app/cases/creator/case-creator.component.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Component, OnInit } from '@angular/core';
-import { Profile } from '../../core/profile/profile.model';
 import { ActivatedRoute } from '@angular/router';
+import { Profile } from '@hmcts/ccd-case-ui-toolkit/dist/shared/profile';
```
Comment (vlaurin) [inline src/app/cases/creator/case-creator.component.ts]: Import seems wrong

File: src/app/core/definitions/definitions.service.spec.ts [hunk 4]
```diff
@@ -3,8 +3,8 @@ import { AppConfig } from '../../app.config';
 import { HttpService } from '@hmcts/ccd-case-ui-toolkit';
 import { Observable } from 'rxjs';
 import { DefinitionsService } from './definitions.service';
-import { CaseType } from '../../shared/domain/definition/case-type.model';
 import createSpyObj = jasmine.createSpyObj;
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
```
Comment (vlaurin) [inline src/app/core/definitions/definitions.service.spec.ts]: Import seems wrong

File: src/app/core/definitions/definitions.service.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { AppConfig } from '../../app.config';
-import { CaseType } from '../../shared/domain/definition/case-type.model';
 import { HttpService } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
```
Comment (vlaurin) [inline src/app/core/definitions/definitions.service.ts]: Import seems wrong

File: src/app/shared/search/filters/search-filters.component.ts [hunk 6]
```diff
@@ -1,12 +1,11 @@
...
-import { CaseState } from '../../domain/definition/case-state.model';
-import { CaseType } from '../../domain/definition/case-type.model';
 import { SearchService } from '../../../core/search/search.service';
 import { SearchInput } from '../../../core/search/search-input.model';
 import { FormGroup } from '@angular/forms';
 import { JurisdictionService } from '../../jurisdiction.service';
-import { OrderService } from '@hmcts/ccd-case-ui-toolkit';
+import { OrderService, Jurisdiction } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
+import { CaseState } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-state.model';
```
Comment (vlaurin) [inline src/app/shared/search/filters/search-filters.component.ts]: 2 imports above seem wrong

File: src/app/shared/search/search-result.component.ts [hunk 7]
```diff
@@ -6,14 +6,13 @@ import { SearchResultViewItemComparator } from './sorting/search-result-view-ite
...
-import { CaseState } from '../domain/definition/case-state.model';
 import { DisplayMode } from '../../core/activity/activity.model';
 import { AppConfig } from '../../app.config';
-import { CaseType } from '../domain/definition/case-type.model';
 import { FormGroup } from '@angular/forms';
 import { ActivityService } from '../../core/activity/activity.service';
-import { CaseReferencePipe, DRAFT_PREFIX } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseReferencePipe, DRAFT_PREFIX, Jurisdiction } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
+import { CaseState } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-state.model';
```
Comment (vlaurin) [inline src/app/shared/search/search-result.component.ts]: 2 imports above seem wrong

File: src/app/workbasket/filters/workbasket-filters.component.ts [hunk 8]
```diff
@@ -1,15 +1,13 @@
...
 import { ActivatedRoute, ActivatedRouteSnapshot, Router } from '@angular/router';
 import { FormGroup } from '@angular/forms';
 import { WorkbasketInputFilterService } from '../workbasket-input-filter.service';
 import { WorkbasketInputModel } from '../workbasket-input.model';
-import { AlertService } from '../../core/alert/alert.service';
 import 'rxjs/add/operator/do';
-import { OrderService } from '@hmcts/ccd-case-ui-toolkit';
+import { OrderService, Jurisdiction, AlertService } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseType } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-type.model';
+import { CaseState } from '@hmcts/ccd-case-ui-toolkit/dist/shared/domain/definition/case-state.model';
```
Comment (vlaurin) [inline src/app/workbasket/filters/workbasket-filters.component.ts]: 2 imports above seem wrong

File: src/app/workbasket/workbasket.component.ts (hunk 2 above)
Comment (andrewfolga) [reply src/app/workbasket/workbasket.component.ts]: Fixed - was not updated after latest release. Visual Studio Code automatically updates it to long path if not refreshed.

File: src/app/cases/creator/case-creator.component.ts (hunk 3 above)
Comment (andrewfolga) [reply src/app/cases/creator/case-creator.component.ts]: Same as above.

File: src/app/shared/search/search-result.component.ts (hunk 7 above)
Comment (andrewfolga) [reply src/app/shared/search/search-result.component.ts]: Done.

File: src/app/shared/search/filters/search-filters.component.ts (hunk 6 above)
Comment (andrewfolga) [reply src/app/shared/search/filters/search-filters.component.ts]: Done

File: src/app/core/definitions/definitions.service.ts (hunk 5 above)
Comment (andrewfolga) [reply src/app/core/definitions/definitions.service.ts]: Done

File: src/app/core/definitions/definitions.service.spec.ts (hunk 4 above)
Comment (andrewfolga) [reply src/app/core/definitions/definitions.service.spec.ts]: Done.

File: src/app/workbasket/filters/workbasket-filters.component.ts (hunk 8 above)
Comment (andrewfolga) [reply src/app/workbasket/filters/workbasket-filters.component.ts]: Done

**Pull request merged**